main.go

```diff
@@ -76,12 +82,17 @@ func saveNotifications(notifs map[string]map[string]Notification, path string) e
 	}
 
 	ioutil.WriteFile(path, rawData.Bytes(), 0600)
+	log.Println("Updated notification cache...")
 	return nil
 }
 
-func addNotification(atTime time.Time, targetUsername string, fromUsername string, threadID string, flowID string) {
-    fmt.Println(notifications)
-	notifications[usernames[targetUsername]][threadID] = Notification{atTime, threadID, flowID, fromUsername}
+func addNotification(notifications map[string]map[string]Notification, atTime time.Time, targetUsername string, fromUsername string, threadID string, flowID string, messageID int64) error {
```

Comment:
I would define a new type e.g. `Notifications` from `map[string]map[string]Notifcation` and thus you can turn `func addNotification(...)` into `func (n Notifications) add(atTime time.Time, ...)`. And it could also be split into its own `notification.go` file.

It would give a better structure to the code, make it easier to read and simpler to test. And you get rid ot all those long `map[string]map[string]Notification` declarations.